react/Gallery.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import GalleryLayout, { GalleryLayoutPropsWithSlots } from './GalleryLayout'
+//@ts-ignore
+import GalleryLegacy from './GalleryLegacy'
+
+export type MobileLayoutMode = 'normal' | 'small' | 'inline'
+
+export interface GalleryLegacyProps {
+  lazyItemsRemaining: number
+  maxItemsPerRow: number
+  minItemWidth: number
+  mobileLayoutMode: MobileLayoutMode
+  products: any[]
+  showingFacets: boolean
+  summary: any
+}
+
+const Gallery: React.FC<
+  GalleryLegacyProps | GalleryLayoutPropsWithSlots
+> = props => {
+  if ('layouts' in props && props.layouts.length > 0) {
+    const {
+      layouts,
+      lazyItemsRemaining,
+      products,
+      showingFacets,
+      summary,
+      ...slots
```

Comment:
This seems a bit strange, if the user passes something else that there isn't a slot it would add to this object as a slot. But I don't know if there is some other way to do this